src/lockfile/parseNpmLock.ts

```diff
@@ -1,6 +1,28 @@
-import { ParsedLock, NpmLockFile } from "./types";
+import { nameAtVersion } from "./nameAtVersion";
+import { ParsedLock, NpmLockFile, LockDependency } from "./types";
 
-export const parseNpmLock = (lock: NpmLockFile): ParsedLock => ({
-  object: lock.dependencies ?? {},
-  type: "success",
-});
+type FormatNpmLock = (
+  previousValue: { },
+  currentValue: [string, LockDependency],
+  currentIndex: number,
+  array: [string, LockDependency][]
+) => { [x: string]: LockDependency; };
+
+/**
+ * formatNpmLock reformats the dependencies object, so the key includes the version, similarly to yarn.lock. For
+ * example, `"@microsoft/task-scheduler": { }` will become `"@microsoft/task-scheduler@2.7.1": { }`.
+ */
+const formatNpmLock: FormatNpmLock = (previousValue, currentValue) => {
+  const [key, dependency] = currentValue;
+  previousValue[nameAtVersion(key, dependency.version)] = dependency;
```

Comment:
One thing that I am not sure about here is that we won’t be adding the range like in `yarn.lock` here. `package.lock` doesn’t include ranges. But since pnpm also doesn’t include ranges, just the installed version, I feel like it should be ok.

Reply:
Yes, this is okay - it's mainly used as a way to tell backfill hasher that "a thing changed" in the hash calculation.